src/pages/home/report/ContextMenu/ContextMenuActions.js

```diff
@@ -96,20 +96,27 @@ export default [
         // the `text` and `icon`
         onPress: (closePopover, {reportAction, selection}) => {
             const message = _.last(lodashGet(reportAction, 'message', [{}]));
-            const html = lodashGet(message, 'html', '');
-
-            const parser = new ExpensiMark();
-            const reportMarkdown = parser.htmlToMarkdown(html);
-
-            const text = selection || reportMarkdown;
+            const messageHtml = lodashGet(message, 'html', '');
 
             const isAttachment = _.has(reportAction, 'isAttachment')
                 ? reportAction.isAttachment
                 : ReportUtils.isReportMessageAttachment(message);
             if (!isAttachment) {
-                Clipboard.setString(text);
+                // This error indicates new code that uses the selection object incorrectly.
+                if (selection && (selection.text || !selection.html)) {
+                    throw new Error('Selection has to either exclusively bear html or be null.');
+                }
+                const resolvedSelection = selection || (messageHtml && {html: messageHtml});
```

Comment:
```suggestion
                const contentToBeCopied = selection || message; 
                if(_.isEmpty(contentToBeCopied)) {
                   return;
                 }
                 	// Save text as HTML when only text is passed for both paste options to work.
                	contentToBeCopied.html = contentToBeCopied.html || contentToBeCopied.text;
                }
```
The message contains both HTML and text. We don't have to explicitly check for HTML, it does not matter.

Reply:
This would require deciding if to produce MD or plaintext in the text field when generating the scraped selection.

Reply:
It won't. I have corrected the previous comment. text will always be text not MD. we will never move the MD around into the text property. Instead convert to MD on demand where needed.